### PR TITLE
ci: upload debuginfo to polar signals

### PR DESCRIPTION
**Pull request by danhhz**
### Motivation

  * This PR adds a feature that has not yet been specified.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by philip-stoev — APPROVED** [review]

**Comment by benesch** [comment]
> updating this to use the new env var

(Is this coming in a future push?)

**Review by benesch — APPROVED** [review]

**Comment by danhhz** [comment]
> > updating this to use the new env var
> 
> (Is this coming in a future push?)

yeah, I was kicking off a ci builder update while typing it. is there a way to set it directly from the i2 thingy or do I just need to re-export the env var from mzbuild.py?

File: misc/python/materialize/mzbuild.py [hunk 1]
```diff
@@ -295,6 +295,23 @@ def run(self) -> None:
                     Tagging={"TagSet": [{"Key": "ephemeral", "Value": ephemeral_str}]},
                 )
 
+            polar_signals_api_token = os.getenv("POLAR_SIGNALS_API_TOKEN")
+            # if self.rd.stable and polar_signals_api_token is not None:
```
Comment (aalexandrov) [inline misc/python/materialize/mzbuild.py:299]: What's the difference between the commented condition and the one that we're actually using now below (that is, what is `self.rd.stable` signifying)?

Comment (danhhz) [reply misc/python/materialize/mzbuild.py]: polar signals charges us for these, so we only upload on tagged version branches

File: ci/builder/Dockerfile [hunk 2]
```diff
@@ -127,7 +127,9 @@ RUN curl -fsSL https://github.com/benesch/autouseradd/releases/download/1.3.0/au
     && curl -fsSL https://github.com/docker/compose/releases/download/v2.15.1/docker-compose-linux-$ARCH_GCC > /usr/local/lib/docker/cli-plugins/docker-compose \
     && chmod +x /usr/local/lib/docker/cli-plugins/docker-compose \
     && curl -fsSL https://github.com/christian-korneck/docker-pushrm/releases/download/v1.9.0/docker-pushrm_linux_$ARCH_GO > /usr/local/lib/docker/cli-plugins/docker-pushrm \
-    && chmod +x /usr/local/lib/docker/cli-plugins/docker-pushrm
+    && chmod +x /usr/local/lib/docker/cli-plugins/docker-pushrm \
+    && curl -fsSL https://github.com/parca-dev/parca-debuginfo/releases/download/v0.7.0/parca-debuginfo_0.7.0_`uname -s`_`uname -m | sed "s/aarch64/arm64/"`.tar.gz \
```
[the remark this answers is not in the document]
Comment (danhhz) [reply ci/builder/Dockerfile]: (done)

File: misc/python/materialize/mzbuild.py (hunk 1 above)
Comment (aalexandrov) [reply misc/python/materialize/mzbuild.py]: So it seems that we want to merge with this condition:

```python
if self.rd.stable and polar_signals_api_token is not None # currently commented out
```

as opposed to this:

```python
if polar_signals_api_token is not None # currently active
```

being enabled, no?

Comment (danhhz) [reply misc/python/materialize/mzbuild.py]: yeah, you were looking at the "DNM testing" commit that I had 😉

**Pull request merged**